"Added artillery, pools, and db routes.

sharmoo opened the pull request:
Removed legacy code.
Added new relic monitoring to test perofmance.
Created full crud functionality for comments and songs"

Commit:
"Added artillery, pools, and db routes.
Removed legacy code.
Added new relic monitoring to test perofmance.
Created full crud functionality for comments and songs"

pafrias commented on Artillery/perf.txt:
```diff
@@ -0,0 +1,291 @@
+    Song CRUD: 13563 (100%)
+  Codes:
+    200: 94941
+    201: 13563
```
I am of the opinion that this entire file should be git ignored

sharmoo replied:
Okay. Done

pafrias commented on Artillery/config.yml:
```diff
@@ -0,0 +1,68 @@
+              id: "{{ commentId }}"
+
+      - delete:
+            url: "/song/{{ songId }}/api/song_id"
```
What songId or range of songIds does this resolve to when testing?

pafrias reviewed (changes requested):
Why are there two index.js files that are both server files?

pafrias commented on server/db.js:
```diff
@@ -87,7 +87,7 @@ const updateSong = (songdata) => {
 
   return new Promise((resolve, reject) => {
-    const query = 'update song set (songname, artistname, aimgurl, hashtag, timeelapsed, thelength, decibel, songurl) = ($2, $3, $4, $5, $6, $7, $8) where id = $1';
+    const query = 'update song set (songname, artistname, aimgurl, hashtag, timeelapsed, thelength, decibel, songurl) = ($2, $3, $4, $5, $6, $7, $8, $9) where id = $1';
```
Could you explain the syntax here? In particular, I'd like to know about what lead you to define a function inside the Promise and then invoke it within the promise.

Bear in mind, I'm not entirely familiar with the client.release() function or using 'try'

pafrias replied:
This may help clarify my curiosity:
https://github.com/petkaantonov/bluebird/wiki/Promise-anti-patterns

pafrias commented on server/.gitignore:
```diff
@@ -0,0 +1,2 @@
+config.js
+newrelic.js
```
Good job 😜

pafrias commented on server/index.js:
```diff
@@ -109,7 +79,7 @@ app.route('/song/:songId/api/song_id')
   //get all comments
   .get((req, res) => {
-    console.log(req.params.songId)
+    //console.log(req.params.songId)
```
Is this the real index.js?

Commit:
"Updates config files.
Refactored database functions in db.js to use promises rather than async/await/etc.
Changed name of database function to readASong.
Updated gitignore and removed legacy code files"

Revert "Adds create, read, update, delete to handle postgres requests for primary and secondary records. Added bodyparser for json handling"

This reverts commit 6cfbb82322c902110bb0df9c92c393a85774cb0a.

Commit:
Amends spacing

pafrias commented on server/db.js:
```diff
@@ -15,182 +16,153 @@ pool.on('error', (err, client) => {
+
 const createSong = (songdata) => {
   // destructure the songdata into its fields
   const { songname, artistname, aimgurl, hashtag, timeelapsed, thelength, decibel, songurl } = songdata;
```
very nice use of object deconstruction

pafrias commented on server/db.js:
```diff
@@ -15,182 +16,153 @@ pool.on('error', (err, client) => {
+const readASong = (songid) => {
+  const query = 'SELECT * from song where id = $1';
+  const params = [songid];
+  return pool.connect()
```
Much cleaner!

pafrias commented on server/db.js:
```diff
@@ -15,182 +16,153 @@ pool.on('error', (err, client) => {
+  const query = 'update song set (songname, artistname, aimgurl, hashtag, timeelapsed, thelength, decibel, songurl) = ($2, $3, $4, $5, $6, $7, $8, $9) where id = $1';
+  return pool.connect()
+    .then(client => {
+      client.query(query, [id, songname, artistname, aimgurl, hashtag, timeelapsed, thelength, decibel, songurl]);
```
should this be returned?

pafrias commented on server/newrelic.js:
```diff
@@ -0,0 +1,53 @@
+'use strict'
```
... oh no....

sharmoo replied:
I'm laughing. And crying. HOW

sharmoo replied:
server/newrelic.js vs newrelic.js
